cmd/internal/kube/daemonset/pods_healthcheck_worker.go

```diff
@@ -0,0 +1,74 @@
+package daemonset
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/postmanlabs/postman-insights-agent/printer"
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (d *Daemonset) checkPodsHealth() {
+	printer.Debugf("Checking pods health, time: %s", time.Now().UTC())
+
+	var podUIDs []types.UID
+	d.PodArgsByNameMap.Range(func(k, _ interface{}) bool {
+		podUIDs = append(podUIDs, k.(types.UID))
+		return true
+	})
+
+	podStatuses, err := d.KubeClient.GetPodsStatusByUIDs(podUIDs)
+	if err != nil {
+		printer.Errorf("Failed to get pods status: %v", err)
+		return
+	}
+
+	for podUID, podStatus := range podStatuses {
+		if podStatus == coreV1.PodSucceeded || podStatus == coreV1.PodFailed {
+			printer.Infof("Pod %s has stopped running", podStatus)
+
+			podArgs, err := d.getPodArgsFromMap(podUID)
+			if err != nil {
+				printer.Errorf("Failed to get podArgs for podUID %s: %v", podUID, err)
```

Comment:
I think this might not be an error.  There can be a change between you calling Range and then running the loop?